catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswQueryFactory.java

```diff
@@ -180,21 +174,11 @@ private CswRecordMapperFilterVisitor buildFilter(QueryConstraintType constraint,
                     CswConstants.NO_APPLICABLE_CODE,
                     null);
         }
-        visitor.setVisitedFilter(filter);
-        boolean isTypeNameCsw = typeNames.contains(new QName(CswConstants.CSW_OUTPUT_SCHEMA,
-                CswConstants.CSW_RECORD_LOCAL_NAME,
-                CswConstants.CSW_NAMESPACE_PREFIX));
-        boolean isTypeNameGmd = typeNames.contains(new QName(GmdMetacardType.GMD_NAMESPACE,
-                GmdMetacardType.GMD_LOCAL_NAME));
-
-        if (isTypeNameCsw || isTypeNameGmd) {
-            try {
-                visitor.setVisitedFilter((Filter) filter.accept(visitor, null));
-            } catch (UnsupportedOperationException ose) {
-                throw new CswException(ose.getMessage(),
-                        CswConstants.INVALID_PARAMETER_VALUE,
-                        null);
-            }
+
+        try {
+            visitor.setVisitedFilter((Filter) filter.accept(visitor, null));
+        } catch (UnsupportedOperationException ose) {
+            throw new CswException(ose.getMessage(), CswConstants.INVALID_PARAMETER_VALUE, null);
```

Comment:
I talked with you initially about this change and between then and the pr ben added in another check to allow another type through. The general though is that for all csw queries we want to transform the csw specific parameters into our internal filter parameters and so we don't need to do this check.